src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -813,9 +813,12 @@ private void SetFailedResult(Status status)
         // 3. There is an existing activity (to enable activity propagation)
         if (diagnosticSourceEnabled || Logger.IsEnabled(LogLevel.Critical) || Activity.Current != null)
         {
-            activity = new Activity(GrpcDiagnostics.ActivityName);
-            activity.AddTag(GrpcDiagnostics.GrpcMethodTagName, Method.FullName);
-            activity.Start();
+            activity = GrpcDiagnostics.ActivitySource.CreateActivity(GrpcDiagnostics.ActivityName, ActivityKind.Internal);
+            if (activity != null)
```

Comment:
do we care about the compatibility scenario here? previously if `diagnosticSourceEnabled` is enabled, the activity creation is forced. Now it is not guaranteed having the Activity gets created. Old listener clients will start miss activities logging.

Reply:
Yes. Noah went into detail: https://github.com/grpc/grpc-dotnet/pull/2244#discussion_r1295028120